### DISCUSSION TITLE
How to access user/item features in sequential model?

### DISCUSSION THREAD

**Discussion by Li-Muyang** [RUCAIBox/RecBole#1053] [Q&A] [answered]
Hi all, I'm trying to access the class of items in ml-100k dataset, however it strikes me that for 2 items belong to a same class (for instance item 32 and  item 48 are both Documentary), they have different class tokens ([7,6,0,0,0,0] and [4,3,7,0,0,0]). Shouldn't two items from same class share same class token? Or did I miscomprehend some where.

**Comment by Ethan-TZ** [comment, marked as answer]
@Li-Muyang Hello, thanks for your attention to RecBole!
In this case, you may use `external_id` to retrieve items.In RecBole, we will `remap` the `item_id`,and use the `internal_id` to index the items
.You can refer to [issue979](https://github.com/RUCAIBox/RecBole/issues/979) for the concepts of `external_id` and `internal_id`.In particular, you can get the information of these two items through the following code:
```
# you can use dataset.token2id to transfer external user token to internal user id
    iid_series = dataset.token2id(dataset.iid_field, ['32', '48'])
    info = dataset.item_feat[iid_series]
```
Their class tokens are [14,0,0,0,0,0].

**Reply by Li-Muyang** [reply]
Thanks, that's very helpful!
